src/utils/activityUtils.ts

```diff
@@ -3,13 +3,35 @@
 *  Licensed under the MIT License. See License.txt in the project root for license information.
 *--------------------------------------------------------------------------------------------*/
 
-import { type ExecuteActivityContext } from "@microsoft/vscode-azext-utils";
+import { ActivityChildType, type ActivityChildItemBase, type ExecuteActivityContext } from "@microsoft/vscode-azext-utils";
 import { ext } from "../extensionVariables";
 import { getWorkspaceSetting } from "../vsCodeConfig/settings";
 
-export async function createActivityContext(): Promise<ExecuteActivityContext> {
+export async function createActivityContext(options?: { withChildren?: boolean }): Promise<ExecuteActivityContext> {
     return {
         registerActivity: async (activity) => ext.rgApi.registerActivity(activity),
         suppressNotification: await getWorkspaceSetting('suppressActivityNotifications', undefined, 'azureResourceGroups'),
+        activityChildren: options?.withChildren ? [] : undefined,
     };
 }
+
+/**
+ * Adds a new activity child after the last info child in the `activityChildren` array.
+ * If no info child already exists, the new child is prepended to the front of the array.
+ * (This utility function is useful for keeping the info children grouped at the front of the list)
+ */
+export function prependOrInsertAfterLastInfoChild(context: Partial<ExecuteActivityContext>, infoChild: ActivityChildItemBase): void {
```

Comment:
I think we should make a universal `activityUtils` in our package by this point.

Reply:
Agreed, I was putting off adding this one because I wasn't sure how often it would be used